Lean4Lean/TypeChecker.lean

```diff
@@ -344,6 +376,7 @@ def whnfCore' (e : Expr) (cheapRec := false) (cheapProj := false) : RecM Expr :=
         pure e
     else
       let r := f.mkAppRevRange 0 rargs.size rargs
+      -- FIXME replace with reduceRecursor? adding arguments can only result in further normalization if the head reduced to a partial recursor application
```

Comment:
Couldn't it also reduce to a lambda?

Reply:
Yes, but it that case it would have went into the first branch of the conditional.